src/NuGet.Core/NuGet.Protocol/CachingSourceProvider.cs

```diff
@@ -67,7 +67,10 @@ public SourceRepository CreateRepository(PackageSource source, FeedType type)
 
         public void AddSourceRepository(SourceRepository source)
         {
-            _cachedSources.TryAdd(source.PackageSource.Source, source);
+            if(_cachedSources.TryAdd(source.PackageSource.Source, source))
```

Comment:
I'm not confident about this change. 

CreateRepository just above does not use the _repositories field. 
In fact the _repositories field only seems to be used during initialization. 

How does this affect later restores? 

Would they end up using this extra source configured?